rpc/types_transaction_receipt.go

```diff
@@ -6,20 +6,21 @@ import (
 	"strconv"
 
 	"github.com/NethermindEth/juno/core/felt"
-	"github.com/NethermindEth/starknet.go/utils"
 )
 
 // CommonTransactionReceipt Common properties for a transaction receipt
 type CommonTransactionReceipt struct {
 	// TransactionHash The hash identifying the transaction
 	TransactionHash *felt.Felt `json:"transaction_hash"`
 	// ActualFee The fee that was charged by the sequencer
-	ActualFee    *felt.Felt       `json:"actual_fee"`
-	Status       TransactionState `json:"status"`
-	BlockHash    *felt.Felt       `json:"block_hash"`
-	BlockNumber  uint64           `json:"block_number"`
-	Type         TransactionType  `json:"type"`
-	MessagesSent []MsgToL1        `json:"messages_sent"`
+	ActualFee       *felt.Felt         `json:"actual_fee"`
+	ExecutionStatus TxnExecutionStatus `json:"execution_status"`
+	FinalityStatus  TxnFinalityStatus  `json:"finality_status"`
+	BlockHash       *felt.Felt         `json:"block_hash"`
+	BlockNumber     uint64             `json:"block_number"`
+	Type            TransactionType    `json:"type"`
+	MessagesSent    []MsgToL1          `json:"messages_sent"`
+	RevertReason    string             `json:"revert_reason"`
```

Comment:
I think `Type` and `RevertReason` are not required fields, you could add _omitempty_ at the end
https://github.com/starkware-libs/starknet-specs/blob/6fadca1f1abf48523c0aa799a10061f40f973468/api/starknet_api_openrpc.json#L2195